### ISSUE TITLE
Authentication not working at all?

### ISSUE THREAD

**Issue by odrotbohm** [jonahwh/tesla-api-client#13]
Looks like the authentication scheme has changed completely? I've tried to adapt the snippet provided in the readme but the authorization request fails with a 400. According to [this website](https://tesla-api.timdorr.com/api-basics/authentication), what is currently implemented is only step 4 albeit with a wrong set of request parameters. :/

I've tried chaning the `grant_type` to `authorization_code`, but that then leads to a 401. That at least indicates an authorization attempt but generally, this feels not working at all.

**Comment by jonahwh** [comment]
Yeah Tesla changed their auth recently - I haven't updated this library because my personal use case allows users to sign in via a Browser.

I can look into implementing the other endpoints, but be aware that because of the occasional captcha that appears, it can be unreliable.

**Comment by odrotbohm** [comment]
Ah, I see. No worries, I just saw a release pretty recently from a bit more than two weeks ago and wondered how that'd be "useful" if the auth is not working. I've used the token generated by Teslafi for my automation, but it looks like Tesla is decreasing the validity of that constantly, and it now has to be refreshed every couple of hours. BIt of a pity that they provide a less convoluted way of obtaining a token.
